File: 55-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int fartest = 0;
        for (int i = 0; i < nums.size() - 1; i++){
            if (i > fartest){
                break;
            }
            if (fartest < (i + nums[i])){
                fartest = i + nums[i];
            }
        }

        if (fartest >= nums.size() - 1){
            return true;
        }
        else {
            return false;
        }

    }
};

File: 55-jump-game.py
from typing import List

nums = [2,3,1,1,4]

class Solution:
    def canJump(self, nums: List[int]) -> bool:
        fartest = 0
        
        for i in range(len(nums) - 1):
            if i > fartest:
                break
            fartest = max(fartest, i + nums[i])
                    
        return True if fartest >= len(nums) - 1 else False
    
sol = Solution()

print(sol.canJump(nums))